src/wallet/wallet.cpp

```diff
@@ -457,8 +457,10 @@ void CWallet::ChainTip(const CBlockIndex *pindex,
 {
     if (added) {
         IncrementNoteWitnesses(pindex, pblock, sproutTree, saplingTree);
+        UpdateSaplingNullifierNoteMapForBlock(pblock);
     } else {
         DecrementNoteWitnesses(pindex);
+        UpdateSaplingNullifierNoteMapForBlock(pblock);
```

Comment:
Nit: If the function isn't being specialised in each case, it could just be called once after the conditionals.

Reply:
Currently not specialised so I can clean this up.